packages/flutter_tools/lib/src/base/analyze_size.dart

```diff
@@ -0,0 +1,336 @@
+    return '--print-instructions-sizes-to=${getAotSizeAnalysisJsonFile(aotOutputPath)}';
+  }
+
+  // Parse the output of unzip -v which shows the zip's contents' compressed sizes.
+  final RegExp _kGetUnzipOutput =
+      RegExp(r'^\s*\d+\s+[\w|:]+\s+(\d+)\s+.*  (.+)$');
+
+  Map<String, dynamic> analyzeApkSize(
+      {@required File apk, @required File aotSizeJson}) {
+    globals.printStatus('▒' * 80);
+    _printEntitySize(
+      '${apk.basename} (total compressed)',
+      apk.lengthSync() ~/ 1024,
+      0,
+    );
+    globals.printStatus('━' * 80);
+    final Directory tempApkContent = globals.fs.systemTempDirectory.createTempSync('flutter_tools.');
+    // TODO: implement Windows.
+    final String unzipOut = processUtils.runSync(<String>[
+      'unzip',
+      '-o',
+      '-v',
+      apk.path,
+      '-d',
+      tempApkContent.path
+    ]).stdout;
+    // We just want the the stdout printout. We don't need the files.
+    tempApkContent.deleteSync(recursive: true);
+
+    final Map<String, dynamic> apkAnalysisJson = _parseUnzipFile(
+      unzipOut,
+      json.decode(aotSizeJson.readAsStringSync()) as List<dynamic>,
+    );
+
+    final List<Map<String, dynamic>> firstLevelPaths = apkAnalysisJson['children'] as List<Map<String, dynamic>>;
+
+    // Print all first level paths and their total sizes.
+    // Expand 'lib' path to print a level deeper.
+    for (final Map<String, dynamic> firstLevelPath in firstLevelPaths) {
+      final String name = firstLevelPath['n'] as String;
+      _printEntitySize(
+        name,
+        (firstLevelPath['value'] as int) ~/ 1024,
+        1,
+      );
+      if (name == 'lib') {
+        _printLibDetails(firstLevelPath, '', aotSizeJson);
+      }
+    }
+
+    globals.printStatus('▒' * 80);
+
+    return apkAnalysisJson;
+  }
+
+  Map<String, dynamic> _parseUnzipFile(
+    String unzipOut,
+    List<dynamic> aotSizeJson,
+  ) {
+    final Map<List<String>, int> pathsToSize = <List<String>, int>{};
+
+    // Parse each path into pathsToSize so that the key is a list of
+    // path parts and the value is the size.
+    // For example:
+    // 'path/to/file' where file = 1500 => pathsToSize[['path', 'to', 'file']] = 1500
+    for (final String line in const LineSplitter().convert(unzipOut)) {
+      final RegExpMatch match = _kGetUnzipOutput.firstMatch(line);
+      if (match == null) {
+        continue;
+      }
+      pathsToSize[match.group(2).split('/')] = int.parse(match.group(1));
+    }
+
+    final Map<String, dynamic> root = <String, dynamic>{'n': '', 'type': 'apk'};
+
+    Map<String, dynamic> currentLevel = root;
+    // Parse through pathsToSize to create a map object with tree-like structure.
+    for (final List<String> paths in pathsToSize.keys) {
+      for (final String path in paths) {
+        if (!currentLevel.containsKey('children')) {
+          currentLevel['children'] = <Map<String, dynamic>>[];
```

Comment:
rather than modifying the json as you go, it would be better to build up an intermediate data structure and then serialize that to json. For example, add a `toJson()` method to that dart object. That will make the code easier for someone else to follow and easier to debug as you will get more benefit from the Dart type system.

Reply:
Done